dm/loader/lightning.go

```diff
@@ -161,25 +197,34 @@ func (l *LightningLoader) restore(ctx context.Context) error {
 			TLS:              cfg.TiDB.TLS,
 		}
 		cfg.Checkpoint.DSN = param.ToDSN()
+		cfg.TiDB.Vars = make(map[string]string)
+		if l.cfg.To.Session != nil {
+			for k, v := range l.cfg.To.Session {
+				cfg.TiDB.Vars[k] = v
+			}
+		}
+		cfg.TiDB.Vars["time_zone"] = "+00:00"
+
 		cfg.TiDB.StrSQLMode = l.cfg.LoaderConfig.SQLMode
 		if err = cfg.Adjust(ctx); err != nil {
 			return err
 		}
-		l.Lock()
-		taskCtx, cancel := context.WithCancel(ctx)
-		l.cancel = cancel
-		l.Unlock()
-		err = l.core.RunOnce(taskCtx, cfg, nil)
+		err = l.runLightning(ctx, cfg)
+		if err == nil {
+			err = lightning.CheckpointRemove(ctx, cfg, "all")
```

Comment:
maybe we can keep the checkpoint in target db and clear it when user start-task with flag `remove-meta`  as syncer do

Reply:
+1. In the correnct dm implementation, we should delay clean checkpoint data when `stop task`

Reply:
we at least should make sure `IsFreshTask` returning `false` after a successful loading, so it needs we don't remove lightning checkpoint or  let `IsFreshTask` inspecting `LightningCheckpointList`